### PR TITLE
[FIX] web: settings view: call server actions with the resId

### PR DESCRIPTION
**Pull request by hubvd**
Steps to reproduce
==================

- Go to settings
- Search for IAP
- Click on View My Services

Cause of the issue
==================

In the settings view, there is a specific hack which removes the resId
from the record

https://github.com/odoo/odoo/blob/e4fe05ac07dbafb993922ff28dc3f7705c639dff/addons/web/static/src/webclient/settings_form_view/settings_form_view.js#L35-L39

This means that the records is not defined in the server action and so
nothing is done

https://github.com/odoo/odoo/blob/cd356f29abcaa1029e8a79ae76c2561f4833232a/addons/iap/views/res_config_settings.xml#L9

Solution
========

Instead of removing the res_id to mark the record as new, override
BasicModel.isNew

opw-3120995

### CODE REVIEW

**Review by jpp-odoo — APPROVED** [review]
LGTM !
Thanks for the fix.

**Review by aab-odoo — APPROVED** [review]

**Comment by aab-odoo** [comment]
robodoo r+

File: addons/web/static/tests/webclient/settings_form_view/settings_form_view_tests.js [hunk 1]
```diff
@@ -1845,4 +1846,52 @@ QUnit.module("SettingsFormView", (hooks) => {
...
+
+        const mockRPC = (route, args) => {
+            if (route === "/web/action/run") {
+                assert.step(route);
+                assert.deepEqual(pick(args.context, "active_id", "active_ids", "active_model"), {
+                    active_id: 1,
+                    active_ids: [1],
+                    active_model: "res.config.settings",
+                });
+                return new Promise(() => {});
```
Comment (kebeclibre) [inline addons/web/static/tests/webclient/settings_form_view/settings_form_view_tests.js:1888]: always pending promise ?

Comment (hubvd) [reply addons/web/static/tests/webclient/settings_form_view/settings_form_view_tests.js:1888]: Is `return Promise.resolve(1);` ok ?

**Pull request closed without merging**